src/util/worker/mod.rs

```diff
@@ -145,22 +153,43 @@ pub struct Worker<T: Display> {
     scheduler: Scheduler<T>,
     receiver: Mutex<Option<Receiver<Option<T>>>>,
     handle: Option<JoinHandle<()>>,
+    timeout: Option<usize>,
+    tasks_per_tick: Option<usize>,
 }
 
-fn poll<R, T>(mut runner: R, rx: Receiver<Option<T>>, counter: Arc<AtomicUsize>, batch_size: usize)
-where
+fn poll<R, T>(
+    mut runner: R,
+    rx: Receiver<Option<T>>,
+    counter: Arc<AtomicUsize>,
+    batch_size: usize,
+    timeout: Option<usize>,
+    tasks_per_tick: Option<usize>,
```

Comment:
tasks_per_tick may conflict with batch_size, maybe they are the same. 

/cc @BusyJay